classes/UpgradeSelfCheck.php

```diff
@@ -318,7 +358,7 @@ public function isOkForUpgrade(): bool
             && ($this->isShopDeactivated() || $this->isLocalEnvironment())
             && $this->isCacheDisabled()
             && $this->isModuleVersionLatest()
-            && $this->isPhpVersionCompatible()
+            && $this->getPhpRequirementsState() != $this::PHP_REQUIREMENTS_INVALID
```

Comment:
```suggestion
            && $this->getPhpRequirementsState() !== $this::PHP_REQUIREMENTS_INVALID
```

Reply:
maybe it's better to type the return of getPhpRequirementsState, right?